src/vma/lwip/tcp_out.c

```diff
@@ -1218,7 +1215,18 @@ tcp_rexmit_segment(struct tcp_pcb *pcb, struct tcp_seg *seg, u32_t wnd)
 
   if (NULL == tcp_split_one_segment(pcb, cur_seg, mss_local, optflags, optlen)) {
       LWIP_DEBUGF(TCP_OUTPUT_DEBUG | 2, ("tcp_split_one_segment: could not allocate memory for segment\n"));
-      return seg;
+      goto out;
+  }
+
+out:
+  /* Update last unsent segment */
+  if (pcb->last_unsent == seg && seg->next != NULL) {
+    for (pcb->last_unsent = cur_seg; pcb->last_unsent->next != NULL;
+         pcb->last_unsent = pcb->last_unsent->next)
+      ;
+#if TCP_OVERSIZE
+    pcb->unsent_oversize = 0;
+#endif /* TCP_OVERSIZE */
```

Comment:
Please consider using `last_unsent` update inside `tcp_split_one_segment()` that might have at least two advantages:
1. no needs in cycle
2. `last_unsent` update is controlled in the function that does splitting.

Proposal:
Call once in function `tcp_split_one_segment()` outside cycle
```
    /* Update last unsent segment */	
    if (pcb->last_unsent == seg) {	
      pcb->last_unsent = cur_seg;	
      pcb->unsent_oversize = oversize;	
    }
```